File: csphn_gifts.cpp
#include<iostream>
#include<vector>

using namespace std;
int a[(int) 1e6 + 5], pref[(int) 1e6 + 5];
int main() {
    int n, k;
    cin >> n >> k;
    for(int i = 1; i <= n; i++) {
        cin >> a[i];
        pref[i] + = 
    }
    int ans = 1e9;
    for(int i = 1; i <= n - k; i++) {
        int tmp = 0;
        for(int j = 1; j <= n - k; j++) {
            if(i >= j + k) {
                tmp = max(tmp, pref[])
            }
            if(i + k >= j) {

            }
        }
        ans = min(ans, tmp1);
    }
    cout << ans;
}

File: csphn_gifts.py
n, k = map(int, input().split())
a = list(map(int, input().split()))
s = [0]*(n+1)
L = [-10**18]*(n+1)
R = [-10**18]*(n+1)
for i in range(1, n+1):
    s[i] = s[i - 1] + a[i - 1]
for i in range(k, n+1):
    L[i] = max(L[i - 1], s[i] - s[i - k])
for i in range(n - k, 0, -1):
        R[i] = max(R[i + 1], s[i + k - 1] - s[i - 1])
res = 10**18
for i in range(k - 1, n - k + 1):
    res = min(res, max(L[i], R[i + k]))
print(res)
